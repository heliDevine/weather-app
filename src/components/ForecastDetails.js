import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
// import WeatherIcon from 'react-icons-weather';

const ForecastDetails = ({ forecast }) => {
    const { date, temperature, humidity, wind } = forecast;
    return (
        <div className="forecast-details" data-testid="forecast-detail">
            <div className="forecast-detail__date">
                {moment(date).format('ddd Do MMM')}
            </div>
            <div className="forecast-details__temperature-max">
                {temperature.max}&deg;C
            </div>
            <div className="forecast-details__temperature-min">
                {temperature.min}&deg;C
            </div>
            <div className="forecast-details__humidity">{humidity}%</div>
            <div className="forecast-details__wind-speed">{wind.speed}mph</div>
            <div className="forecast-details__wind-direction">
                wind direction:
                {wind.direction}
                {/* <WeatherIcon name="s" /> */}
            </div>
        </div>
    );
};

export default ForecastDetails;

ForecastDetails.propTypes = {
    forecast: PropTypes.shape({
        date: PropTypes.number.isRequired,
        humidity: PropTypes.number.isRequired,
        temperature: PropTypes.shape({
            min: PropTypes.number,
            max: PropTypes.number,
        }).isRequired,
        wind: PropTypes.shape({
            speed: PropTypes.number,
            direction: PropTypes.string,
        }).isRequired,
    }).isRequired,
};
