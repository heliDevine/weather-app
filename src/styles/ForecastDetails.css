.forecast-details {
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 10px, 0, 0, 0;
    align-items: center;
    margin: 20px;
}
.forecast-detail__date {
    font-size: 20px;
    font-weight: 900;
    padding: 5px;
}
.forecast-details__temperature-max {
    font-weight: 900;
}
