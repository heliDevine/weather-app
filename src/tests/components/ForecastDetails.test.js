import React from 'react';
import { render } from '@testing-library/react';
import ForecastDetails from '../../components/ForecastDetails';
import { number } from 'prop-types';

describe('ForecastDetails', () => {
    const validProps = {
        date: 1111111,
        temperature: {
            min: 12,
            max: 22,
        },
        wind: {
            speed: 60,
            direction: 'nne',
        },
        humidity: 50,
    };

    it('renders correctly', () => {
        const { asFragment } = render(
            <ForecastDetails
                date={validProps.date}
                temperature={validProps.temperature}
                wind={validProps.wind}
                humidity={validProps.humidity}
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
it('renders correct values for props', () => {
    const { getByText, getByTestId } = render(
        <ForecastDetails
            date={validProps.date}
            temperature={validProps.temperature}
            wind={validProps.wind}
            humidity={validProps.humidity}
        />,
    );
    expect(getByText('Thu 1st Jan')).toHaveClass('forecast-detail__date');
    expect(getByTestId('forecast-icon')).toHaveClass('forecast-summary__icon');
    expect(getByText('22Â°C')).toHaveClass('forecast-details__temperature-max');
});
