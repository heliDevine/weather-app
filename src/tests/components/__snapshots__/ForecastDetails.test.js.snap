// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<div
  className="forecast-details"
  data-testid="forecast-detail"
>
  <div
    className="forecast-detail__date"
  >
    Thu 1st Jan
  </div>
  <div
    className="forecast-details__temperature-max"
  >
    22
     °C
  </div>
  <div
    className="forecast-details__temperature-min"
  >
    12
     °C
  </div>
  <div
    className="forecast-details__humidity"
  >
    humidity: 
    50
    %
  </div>
  <div
    className="forecast-details__wind-speed"
  >
    wind speed: 
    60
    mph
  </div>
  <div
    className="forecast-details__wind-direction"
  >
    wind direction: 
    nne
  </div>
</div>
`;
